docs/reference/modules/datatiers.asciidoc

```diff
@@ -0,0 +1,104 @@
+[role="xpack"]
+[[modules-tiers]]
+=== Data tiers
+
+Common data lifecycle management patterns revolve around transitioning the indices
+through multiple collections of nodes with different hardware characteristics in order
+to fulfil evolving CRUD, search, and aggregation needs as the indices age. The concept
+of a tiered hardware architecture is not new in {es}.
+<<index-lifecycle-management, Index Lifecycle Management>> was instrumental in
+implementing tiered architectures by automating the managemnt of indices according to
+performance, resiliency and data retention requirements.
+The <<overview-index-lifecycle-management, hot-warm-cold>> architectures are common
+for timeseries data such as logging and metrics. However without a formalised `tier`
+concept in {es} the hot-warm-cold architecture implementation was based on user specified
+node attributes used in conjunction with <<shard-allocation-awareness, shard allocation awareness>>.
+
+A data tier is a collection of nodes with the same role. Data tiers are now an integrated
+solution offering better support for optimising cost and improving performance.
+Each tier is formalising specific characteristics and data behaviours.
+
+The node roles that can currently define data tiers are:
+
+* <<data-content-node, data_content>>
+* <<data-hot-node, data_hot>>
+* <<data-warm-node, data_warm>>
+* <<data-cold-node, data_cold>>
+
+The more generic <<data-node, data role>> is not a data tier role, but
+it is the default node role if no roles are configured. If a node has the
+<<data-node, data>> role we will treat the node as if it has all of the tier
+roles assigned.
+
+[[content-tier]]
+==== Content tier
+
+The content tier is made of one or more nodes that have the <<data-content-node, data_content>>
+role. A content tier is designed to store and search user created content. Non-timeseries data
+doesn't necessarily follow the hot-warm-cold path. The hardware profiles are quite different to
+the <<hot-tier, hot tier>>. User created content will prioritise high CPU to support complex
+queries and aggregations in a timely manner, as opposed to the <<hot-tier, hot tier>> which
+will prioritise high IO.
+The content tier is a fast, higher cost, tier. The content data has very long data retention
+characteristics and from a resiliency perspective the indices in this tier use one or more
+replicas.
```

Comment:
Thanks Lee, I struggled with this a bit and thought it's an important motivation for data tiers and should be included in the docs in order to illustrate the value and a way to reason about tiers. 

I understand your concerns but I do wonder if this kind of information would still have a place in our documentation in the form or recommendations/vision/purpose for each tier? (and underline the users can still place whatever nodes they want in each tier)

I'm otherwise a bit confused as to what sort of information should we provide for each tier, besides that it's a set of nodes with a particular role.